akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -358,6 +358,22 @@ trait PathMatchers {
 
   /**
    * A PathMatcher that matches and extracts the complete remaining,
+   * unmatched part of the request's URI path as an decoded (with actual whitespace) String.
+   * @group pathmatcher
+   */
+  object RemainingDecoded extends PathMatcher1[String] {
+    private val encodedSpace = "%20"
+    private val decodedSpace = " "
+
+    def decodeWhitespace(path: String) = path.replace(encodedSpace, decodedSpace)
```

Comment:
Thanks for your comment. Now I do see that my implementation was naive (at best). Can you please advise on how to access segments of the path? I was trying to do something like this:
 `RemainingPath(path).extractions.join(())` 
however it provides tuple with single element which is encoded with `%20`.